cmd/execution/main.go

```diff
@@ -161,6 +176,36 @@ func main() {
 			pendingBlocks = buffer.NewPendingBlocks() // for following main chain consensus
 			return nil
 		}).
+		Component("Block data uploader", func(builder cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+			if enableBlockDataUpload {
+
+				logger := node.Logger.With().Str("component_name", "block_data_uploader").Logger()
+				gcpBucketUploader, err := uploader.NewGCPBucketUploader(
+					context.Background(),
+					gcpBucketName,
+					logger,
+				)
+				if err != nil {
+					return nil, fmt.Errorf("cannot create GCP Bucket uploader: %w", err)
+				}
+
+				asyncUploader := uploader.NewAsyncUploader(
+					gcpBucketUploader,
+					logger,
+					collector,
+				)
+
+				blockDataUploader = asyncUploader
```

Comment:
Do we want to use the async uploader by default? It never returns any error, which means that the computation manager will not know about any errors that occur.

In general, this means that if even a single upload fails, we will never try again and DPS would get stuck at that point because that the result for that block would never be uploaded right?

Do we want to build in some retry mechanism for the uploader so it can hopefully succeed to upload blocks in the face of transient failures?

Reply:
I'm adding some retry logic, however we cannot rely on this for too much - don't want to have queue backs up and destabilise EN operation.
I think this is the very first version, in subsequent we would want ability for multiple EN to write to same bucket.

Reply:
Sure, I think it would make sense to retry fewer times, and emit some metric if in the end all retries failed.